### PR TITLE
Fragment base component #132

### LINKED ISSUE #138: Fragment selector + usage in base part

### LINKED ISSUE #139: Fragment preview handling

### PR DESCRIPTION
**Pull request by pmi**
Fragment selector + usage in base part (closes #138)
Fragment preview handling (closes #139)

- get rid of pageAsJson completely
- create region structure from components array only

### CODE REVIEW

File: src/cms/queries/_getMetaData.ts [hunk 1]
```diff
@@ -29,14 +28,27 @@ export const PAGE_FRAGMENT = `
             imageUrl (type:absolute, scale: "width-768")
           }
         }
+`
+
+// THIS QUERY DOES NOT SUPPORT NESTED FRAGMENTS
```
Comment (espen42) [inline src/cms/queries/_getMetaData.ts:33]: Not a change request, just curious: should it? Will it later? Will another query be able to?

File: src/cms/typesRegistration.ts [hunk 2]
```diff
@@ -38,6 +39,10 @@ TypesRegistry.addContentType(MOVIE_CONTENTTYPE_NAME, {
     view: MovieView,
 });
 
+TypesRegistry.addContentType(FRAGMENT_CONTENTTYPE_NAME, {
```
Comment (espen42) [inline src/cms/typesRegistration.ts:42]: Move inside xpAdapter/TypeRegistry.ts at some point?

File: src/cms/typesRegistration.ts [hunk 3]
```diff
@@ -52,6 +57,10 @@ TypesRegistry.addComponent(XP_COMPONENT_TYPE.LAYOUT, {
     view: BaseLayout
 });
 
+TypesRegistry.addComponent(XP_COMPONENT_TYPE.FRAGMENT, {
```
Comment (espen42) [inline src/cms/typesRegistration.ts:60]: Same with these, incl. `.IMAGE`, `.PART`, `.LAYOUT` and `.TEXT` - move inside xpAdapter/TypeRegistry.ts at some point?

**Review by espen42 — APPROVED** [review]

File: src/cms/queries/_getMetaData.ts (hunk 1 above)
Comment (pmi) [reply src/cms/queries/_getMetaData.ts:33]: Well, it's just a memo to self ;) 
Only top level fragments do `components(resolveFragments: false)` now, but it is possible to add it to fragments' inner components queries as well

File: src/cms/typesRegistration.ts (hunk 2 above)
Comment (pmi) [reply src/cms/typesRegistration.ts:42]: Yep, in scope of better adapter/cms separation task

File: src/cms/typesRegistration.ts (hunk 3 above)
Comment (pmi) [reply src/cms/typesRegistration.ts:60]: Agree

**Pull request merged**